firefox-ios/Client/Application/RemoteSettings/RemoteDataType.swift

```diff
@@ -20,30 +20,62 @@ enum RemoteDataTypeError: Error, LocalizedError {
 
 enum RemoteDataType: String, Codable {
     case passwordRules
+    case contentBlockingLists
 
     var type: any RemoteDataTypeRecord.Type {
         switch self {
         case .passwordRules:
             return PasswordRuleRecord.self
+        case .contentBlockingLists:
+            return ContentBlockingListRecord.self
         }
     }
 
-    var fileName: String {
+    var fileNames: [String] {
         switch self {
         case .passwordRules:
-            return "RemotePasswordRules"
+            return ["RemotePasswordRules"]
+        case .contentBlockingLists:
+            return BlocklistFileName.allCases.map { $0.filename }
         }
     }
 
     var name: String {
         switch self {
         case .passwordRules:
             return "Password Rules"
+        case .contentBlockingLists:
+            return "Content Blocking Lists"
         }
     }
 
+    /// Loads the local settings for the given data type record, returning the
+    /// decoded objects.
+    /// - Returns: settings decoded to their RemoteDataTypeRecord.
```

Comment:
Eventually I think we would want to converge to the desktop implementation where a `record` and `attachment` is a one to one mapping. So the API might look like `record.get_attachment`. But this also works.

Reply:
@issammani Once this lands, I can add a patch for you two to review the attachment portion where yes you can do some part in python however locally it should all work out in terms of record.getAttachment(...name)